### PR TITLE
Mm 1640 Fixed odd scrolling issues on channel refresh

### PR DESCRIPTION
**Pull request by nickago**
This is a re-implementation of how our scrolling worked. The old method had become too organic with edge cases growing out of it, this redesigns it in a simple and easily extensible manner. 

Note: I have removed perfectScrollbar from the main pane. This was due to the very unpredictable behavior which it would induce in my scrolling tests.

### CODE REVIEW

File: web/react/components/post_list.jsx [hunk 1]
```diff
@@ -55,44 +57,23 @@ module.exports = React.createClass({
...
-        $(window).resize(function(){
-            $(post_holder).perfectScrollbar('update');
-
-            // this only kind of works, detecting zoom in browsers is a nightmare
-            var newZoom = (window.outerWidth - 8) / window.innerWidth;
-
-            if (self.scrollPosition >= post_holder.scrollHeight || (self.oldScrollHeight != post_holder.scrollHeight && self.scrollPosition >= self.oldScrollHeight) || self.oldZoom != newZoom) self.resize();
+        // Timeout exists for the DOM to fully render before making changes
+        var self = this
+        setTimeout(function initialScroll() {
```
Comment (crspeller) [inline web/react/components/post_list.jsx]: This seems hacky. A quick google reveals that "DOMContentLoaded" or a JQuery alternative my be a possibility. Why not those?

**Comment by crspeller** [comment]
Many style guide violations in this. Did you forget to run ESLint?

**Comment by nickago** [comment]
No, I figured for a more substantial change like this, I would get the commit in and then do a PR for refactoring, just because it's a large refactor and I don't want to obfuscate the review. However, I add a refactor commit now

**Comment by nickago** [comment]
Ready for re-review @crspeller

**Comment by crspeller** [comment]
Looks good. Need a seconder.

**Comment by jwilander** [comment]
+1

**Pull request merged**